Adding get drivers()

Exoutia opened the pull request:
<!--
Thank you for contributing to F1PyStats!
Kindly fill out the following sections below to submit a pull request.
-->

## Description
<!-- Briefly describe the changes (additions/deletions) you have made to the code, and why-->
- I have added the get_drivers function with option to add a round numbers as parameters. 
- This function will return the a panda df with name, permanent number, nationality and date of birth column.
- As there is no permanent number before 2014 (according to api documentation) the function will return none in permanent number column if year is less than 2014 in parameter

## Related Issue(s)
<!-- Please note any issue numbers this pull request addresses/resolves (There should be at least one)-->
Fixes #41 <!--Replace X with this pull request's related issue number-->

## Screenshots (If necessary)
<!-- Include screenshots here, if you have made any visual changes to the application -->
## for 2014
![image](https://user-images.githubusercontent.com/94464106/193771160-595ba38f-3dd0-460d-97c0-3fdb4d8a4296.png)
## below 2014
![image](https://user-images.githubusercontent.com/94464106/193771371-b8ec9c38-2932-4c93-a5c5-baf1697d53c1.png)

alec-kr commented:
Great work 💯 

However, Pylint is not happy with some of the code format. 

![image](https://user-images.githubusercontent.com/52685467/193842931-e12fc615-21d5-480c-8afc-1c4080f77f15.png)

- You must remove the trailing whitespaces from lines 6, 10, 14, 18 and 21 of ```driver_info.py```
- You must also add a class docstring to the DriverInfo class. Great work on adding the other docstrings btw ;)
- There are also a few trailing whitespaces in ```f1pystats.py``` at lines 33, 41, 43 and 47.

Your PR should be able to merge after the lint issues have been fixed :)

alec-kr reviewed (changes requested):
Pylint is not happy with some of the code format. 

![image](https://user-images.githubusercontent.com/52685467/193842931-e12fc615-21d5-480c-8afc-1c4080f77f15.png)

- You must remove the trailing whitespaces from lines 6, 10, 14, 18 and 21 of ```driver_info.py```
- You must also add a class docstring to the DriverInfo class. Great work on adding the other docstrings btw ;)
- There are also a few trailing whitespaces in ```f1pystats.py``` at lines 33, 41, 43 and 47.

Your PR should be able to merge after the lint issues and README have been fixed :)

Exoutia commented:
Ok I will do as you pointed out this is my first serious code contribution 😁😁 in open source I will fix them asap

alec-kr commented:
Great work on fixing the linter issues.

Don't forget to update the README and add your test for the new module. (See my [comment](https://github.com/alec-kr/F1PyStats/pull/45#issuecomment-1267079213), in case you missed it)

Exoutia commented:
> Great work on fixing the linter issues.
> 
> Don't forget to update the README and add your test for the new module. (See my [comment](https://github.com/alec-kr/F1PyStats/pull/45#issuecomment-1267079213), in case you missed it)

I added the test module and updated the read me

alec-kr commented on tests/test_driver_info.py:
```diff
@@ -0,0 +1,35 @@
+    
+    def test_driver_names(self):
+        """Test for the get_driver_names methods of DriverInfo"""
+        assert self.driver.get_driver_names() == ['Jean Alesi', 'Rubens Barrichello', 'Luciano Burti']
```
Your test file calls ```get_driver_names()```, but there is no such function in ```DriverInfo```.

```DriverInfo``` contains a function called ```get_drivers_names()```. (NOTE the difference with ```drivers``` and ```driver```)

alec-kr commented on tests/test_driver_info.py:
```diff
@@ -0,0 +1,35 @@
+    
+    def test_driver_nationality(self):
+        """Test for the get_driver_nationality method of DriverInfo"""
+        assert self.driver.get_driver_nationality() == ['French', 'Brazilian', 'Brazilian']
```
You are calling ```get_driver_nationality()```, but there is no such function in ```DriverInfo```.

```DriverInfo``` contains ```get_drivers_nationality()```. (NOTE the difference with ```drivers``` and ```driver```)

alec-kr replied:
You should determine if you will use ```get_driver...``` or ```get_drivers...``` prefix for both test functions.

Exoutia commented:
Sorry for these small mistake I was doing these changes at 1 AM in my timezone so I think i was sleepy 🥲🥲 didn't notice them and in my jupyter note I was mostly relying on auto complete so I didn't find any error during my test run, again sorry I have made those changes as you have suggested.

alec-kr commented:
> Sorry for these small mistake

It's okay. You can't always get everything right on the first try :)

alec-kr reviewed (changes requested):
There are some changes that need to be made to pass the Pylint checks:

![image](https://user-images.githubusercontent.com/52685467/193974810-3802b5c1-6ef5-40e0-918c-c4f3e116c68e.png)

Exoutia commented:
Maintainer works seems very tiring :cry:

alec-kr commented:
> Maintainer works seems very tiring

This is the first time I've been maintaining a project. It can be tiring at times.

> thanks for doing the hard work

Thank you for contributing 😁
